### PR TITLE
improved file existence checking; use braced vars in strings

### PR DESCRIPTION
**Pull request by TheDigitalOrchard**
Improved file existence checking; use braced vars in strings for improved readability.

- bug fix / new feature?   
- BC break? no
- doc PR: nette/docs#???

### CODE REVIEW

File: src/Latte/Engine.php [hunk 1]
```diff
@@ -215,7 +215,7 @@ public function getCacheFile(string $name): string
 			? preg_replace('#[^\w@.-]+#', '-', substr($m[0], 1)) . '--'
 			: '';
 		$file = "{$this->tempDirectory}/{$base}{$hash}.php";
-		return file_exists($file) ? $file : false;
```
Comment (milo) [inline src/Latte/Engine.php]: IMHO,  `null` and `?string` return type is better.

Comment (TheDigitalOrchard) [reply src/Latte/Engine.php]: What are your thoughts on checking for the existence of the file before trying to read the file?

I'm surprised by the use of the silence @ operator throughout, and in my enviro (and likely some others), it still generates errors due to how the framework in use handles such errors, even when silenced.

Comment (milo) [reply src/Latte/Engine.php]: Sorry, I overlooked that you added `file_exists()` code. That's wrong. Method only builds the cache file name.

**Comment by dg** [comment]
Whats wrong with current solution? It is really elaborated peace of code.

**Comment by milo** [comment]
Closed for reasons written above.

You can't prevent race conditions without exclusive locking, which is quite expensive for some cases, so usage of `@` operator is appropriate. For example, code proposed by you:
```php
file_exists($file) ? (int)filemtime($file) : 0
```
may throw warning, when `$file` is deleted between `file_exist()` and `filemtime()` calls.

**Pull request closed without merging**